config/fi_provider.m4

```diff
@@ -59,6 +59,11 @@ AC_DEFUN([FI_PROVIDER_SETUP],[
 	      [],
 	      [enable_$1=auto])
 
+	# Save CPPFLAGS and LDFLAGS before they are modified by FI_CHECK_PREFIX_DIR
+	# Provider's local macros could use the value if needed.
+	$1_orig_CPPFLAGS="$CPPFLAGS"
+	$1_orig_LDFLAGS="$LDFLAGS"
```

Comment:
Fun fact: when assigning from one shell variable to another, you don't need quotes.  I.e., the following is sufficient (and just ever-so-slightly more efficient):

```sh
$1_orig_LDFLAGS=$LDFLAGS
```

(yes, this works even if `$LDFLAGS` contains spaces)